### ISSUE TITLE
typeahead.jquery.js: better change module name from "typeahead.js" to 'typeahead"

### ISSUE THREAD

**Issue by PetersonLian** [twitter/typeahead.js#1438]
```
(function(root, factory) {
    if (typeof define === "function" && define.amd) {
    define("typeahead.js", [ "jquery" ], function(a0) {
        return factory(a0);
    });
  } else if (typeof exports === "object") {
    module.exports = factory(require("jquery"));
  } else {
    factory(jQuery);
  }
})(this, function($) {
...
```

in code snippet `define("typeahead.js", ["jquery"], function(a0){...`, better change `"typeahead.js"` to `"typeahead"`.

**Comment by myxibrium** [comment]
I agree. I couldn't figure out how to get this plugin to work using Require.js without changing that line of code to remove the ".js" at the end.

I saw other people having similar issues, not sure if they are all related.

https://github.com/twitter/typeahead.js/issues/1211
https://github.com/bassjobsen/Bootstrap-3-Typeahead/issues/106
https://github.com/twitter/typeahead.js/issues/1271

**Comment by micahblu** [comment]
I am having the same issue, I have to manually change typehead.js to typehead in order for it to work as an commonjs module. Should be a very quick and easy fix

**Comment by micahblu** [comment]
I've submitted a pr for this: https://github.com/twitter/typeahead.js/pull/1465

**Comment by yanickrochon** [comment]
This should be fixed. I just encountered this too.

**Comment by micahblu** [comment]
I think the project is dead guys :/

**Comment by yanickrochon** [comment]
Actually, it's being partly maintained [here](https://github.com/corejavascript/typeahead.js).

And I opened a similar issue, [here](https://github.com/corejavascript/typeahead.js/issues/66).

**Comment by nikcub** [comment]
i'm maintaining a fork here:

https://github.com/nikcub/typeahead

which is registered on bower as `typeahead-requirejs`
